[Node][Flaky Test] Container tests - Node 26.x libatomic.so.1 missing on Amazon Linux

### Test Name

Node container tests (all) - libatomic.so.1 missing

### Test Location

Container build environment setup (amazon-linux image)

### Failure Permlink

https://github.com/valkey-io/valkey-glide/actions/runs/28494732381/job/84458567398#logs

### Frequency

Observed in scheduled CI run on 2026-07-01. Failed in all 6 container test jobs (Node 26.x across valkey 7.2, 8.0, 8.1, 9.0, redis 6.2, 7.0).

### Steps to Reproduce

1. Run the FMT Node workflow on schedule
2. Observe failure in all `test-node-container` jobs using Node 26.x on amazon-linux

### System Information

GitHub Actions (amazon-linux container, ubuntu-latest host, x64, self-hosted)

### Language and Version

Node.js 26.x

### Engine Version

All engine versions affected (valkey 7.2, 8.0, 8.1, 9.0, redis 6.2, 7.0)

### Logs

```
[warning]/__t/node/26.4.0/x64/bin/node: error while loading shared libraries: libatomic.so.1: cannot open shared object file: No such file or directory

node: error while loading shared libraries: libatomic.so.1: cannot open shared object file: No such file or directory

##[error]Process completed with exit code 127.
```

### Screenshots

_No response_

### Glide Version

main

### Expected Behavior

Node.js 26.x should run successfully in the Amazon Linux container environment.

### Actual Behavior

Node.js 26.4.0 requires `libatomic.so.1` which is not installed in the Amazon Linux container image. This causes all container-based Node tests to fail immediately with exit code 127 (command not found / shared library missing).

### Possible Fixes

Add `libatomic` to the container setup dependencies. For Amazon Linux 2023, add `yum install -y libatomic` or `dnf install -y libatomic` to the container initialization steps. This is likely a new dependency introduced by Node.js 26.x that was not required by earlier Node versions.

--- a/.github/actions/install-shared-dependencies/action.yml
+++ b/.github/actions/install-shared-dependencies/action.yml
@@ -49,7 +49,7 @@ runs:
           shell: bash
           if: "${{ inputs.os == 'amazon-linux' }}"
           run: |
-              yum install -y gcc pkgconfig openssl openssl-devel which curl gettext libasan tar --allowerasing
+              yum install -y gcc pkgconfig openssl openssl-devel which curl gettext libasan libatomic tar --allowerasing
 
         - name: Setup WSL (Windows only)
           if: "${{ inputs.os == 'windows' && inputs.engine-version }}"

